client/components/CompanyList.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react'
+import {connect} from 'react-redux'
+
+import {getCompanies} from '../actions/companies'
+
+class CompanyList extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      search: '',
+      companyNames: []
+    }
+    this.updateSearch = this.updateSearch.bind(this)
+    this.searchBar = this.searchBar.bind(this)
+  }
+  componentDidMount () {
+    this.props.dispatch(getCompanies())
+      .then(() => {
+        this.setState({
+          companyNames: this.props.companyList.map(company => company.name)
+        })
+      })
+  }
+
+  searchBar (e) {
+    if (e.target.value === '') {
+      this.setState({
+        companyNames: this.props.companyList.map(company => company.name)
+      })
+      this.setState({
+        search: e.target.value
+      })
+    } else {
+      let searchList = this.props.companyList.map(company => company.name)
+      searchList = searchList.filter(res => {
+        return res.toLowerCase().search(e.target.value.toLowerCase()) !== -1
+      })
+      this.setState({
+        search: e.target.value
+      })
+      this.setState({
+        companyNames: searchList
+      })
+    }
+  }
+
+  updateSearch (e) {
+    this.setState({
```

Comment:
I don't think you need this function. Remove if you don't need it.

Reply:
Done